pandas/io/formats/info.py

```diff
@@ -130,41 +137,58 @@ def info(
     ids, dtypes = _get_ids_and_dtypes(data)
     col_count = len(ids)
 
-    if col_count == 0:
+    if col_count == 0 and isinstance(data, ABCDataFrame):
```

Comment:
can you try to avoid doing if DataFrame else in this implementation. It makes it almost impossible to read.
alternatively you can provide a base class, with DataFrame and Series subclasses to handle the formatting. I am not sure which is easier / cleaner.

Reply:
@jreback have factored out `mem_usage` and `counts` as well, and also reduced branching in the main body.

There's still some branching though - is this OK, or would you prefer a base class with Frame/Series subclasses?